examples/ring_topology.rs

```diff
@@ -0,0 +1,42 @@
+use rr_channel::detthread;
+use rr_channel::crossbeam;
+use std::sync::Mutex;
+use std::sync::Arc;
+
+fn main() {
+    let mut senders: Vec<Mutex<_>> = Vec::new();
+    let mut receivers: Vec<Mutex<_>> = Vec::new();
+
+    for _ in 0..10 {
+        let (s, r) = crossbeam::unbounded::<i32>();
+        senders.push(Mutex::new(s));
+        receivers.push(Mutex::new(r));
+    }
+    
+    let senders = Arc::new(senders);
+    let receivers = Arc::new(receivers);
+
+    for i in 0..10 {
+        // Clone via arc for moving into closure.
+        let senders = senders.clone();
+        let receivers = receivers.clone();
+        rr_channel::detthread::spawn(move || {
```

Comment:
We should actually collect the handles from the spawned threads and wait for them before exiting the main thread. Otherwise not all threads may get to execute. You can see an example of collecting handles and waiting for them here: https://github.com/gatoWololo/rr_channel/issues/38#issue-616258553